pyHS100/pyHS100.py

```diff
@@ -54,12 +59,40 @@ def __init__(self, ip_address):
         """
         Create a new SmartPlug instance, identified through its IP address.
 
-        :param ip_address: ip address on which the device listens
+        :param str ip_address: ip address on which the device listens
+        :raises SmartPlugException: when unable to communicate with the device
         """
         socket.inet_pton(socket.AF_INET, ip_address)
         self.ip_address = ip_address
 
-        self.alias, self.model, self.features = self.identify()
+        self.sys_info = self.get_sysinfo()
+
+        self._alias, self.model, self.features = self.identify()
+
+    def _query_helper(self, target, cmd, arg={}):
+        """
+        Query helper, returns unwrapped result object.
+        Raises SmartPlugException in case of error reported by the plug.
+
+        :param target: Target system {system, time, emeter, ..}
+        :param cmd: Command to execute
+        :param arg: JSON object passed as parameter to the command, defualts to {}
+        :return: Unwrapped result for the call.
+        :rtype: dict
+        :raises SmartPlugException: if command was not executed correctly
+        """
+        response = TPLinkSmartHomeProtocol.query(
+            host=self.ip_address,
+            request={target: {cmd: arg}}
+        )
+
+        result = response[target][cmd]
+        if result["err_code"] != 0:
+            raise SmartPlugException("Error on {}.{}: {}".format(target, cmd, result))
```

Comment:
I'd prefer to keep this again for the sake of readability. Moving the call to format to a new line or splitting up the parameter list doesn't really improve reaadability.